gym_cap/envs/agent.py

```diff
@@ -229,6 +230,62 @@ def get_loc(self):
     def report_loc(self):
         print("report: position x:%d, y:%d" % (self.x, self.y))
 
+    def get_obs(self, com_ground, com_air, env, *args):
+        if self.team == BLUE:
+            myTeam = env.get_team_blue()
+        else:
+            myTeam = env.get_team_red()
+
+        a = 39              # env.map_size[0]*2-1
+        b = 39              # env.map_size[1]*2-1
+        obs = np.full(shape = (a, b), fill_value = OBSTACLE)
+        val = env.get_full_state()
+
+        loc = self.get_loc()
+        targetx, targety = loc[0], loc[1]
+
+        for i in range(a-targetx-1, 2*a-targetx-1):
```

Comment:
https://github.com/raide-project/ctf_public/blob/09460d70f0a4e7c9402dacfa2f9f24d84759fe37/gym_cap/envs/cap_env.py#L161-L168

Try using this code for making the vision within a certain range.